scalding-core/src/main/scala/com/twitter/scalding/StringUtility.scala

```diff
@@ -0,0 +1,39 @@
+package com.twitter.scalding
+
+object StringUtility {
+  // Scala's split function uses regex which is too slow thus we create this faster function
+  private def isTrailingWith(text: String, separator: String) = {
+    val result = text.foldRight((List[Boolean](), true)) {
+      case (c, (tillNow, isTrailingBefore)) =>
+        val trailing = isTrailingBefore && (c.toString == separator)
+        (tillNow ++ List(trailing), trailing)
+    }
+    result._1.reverse
+  }
+
+  private def fastSplitHelper(text: String, key: String, isTrailingAt: List[Boolean]): Seq[String] = {
+    if (text.isEmpty) {
+      Seq()
+    } else {
+      val firstIndex = text.indexOf(key)
+      if (firstIndex == -1) {
```

Comment:
this is probably the common case. I'd like to see us optimize for this case so in that case our only call is text.indexOf(key).

Also, why isn't something like this good enough:

```scala
def fastSplit(s: String, sep: String, from: Int): List[String] = {
  val idx = s.indexOf(sep, from)
  if (idx == -1) List(s)
  else {
    (s.substring(from, idx)) :: fastSplit(s, sep, idx + sep.length)
  }
}
```
Seems like something like that is going to be faster and simpler.